packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -504,7 +504,26 @@ export function typeDefinition<T>(
 		}
 	}
 
-	// TODO iterate final tree and call `options.attachTypeInfo`
+	// TODO: attach a combination of all possible definitions
+	const attached = new Set<T>()
+
+	function attachTypeInfo(node: CheckerTreeRuntimeNode<T>) {
+		for (const def of node.validDefinitions) {
+			if (!attached.has(node.node.originalNode)) {
+				options.attachTypeInfo(node.node.originalNode, def.typeDef)
+				attached.add(node.node.originalNode)
+			}
+		}
+		for (const child of node.children) {
+			for (const node of child.possibleValues) {
+				attachTypeInfo(node)
+			}
+		}
+	}
```

Comment:
The set and TODO shouldn't really be neccessary anymore.

It should be fine to attach a union type containing all `node.validDefinitions`.

Reply:
Cool! I'll try working on this

Reply:
Ah shit 😄 

![image](https://github.com/SpyglassMC/Spyglass/assets/17352009/20907a5b-bfef-4ad1-962e-7ce20964b306)